### ISSUE TITLE
Simple enum support

### ISSUE THREAD

**Issue by hakuzumon** [EvidentSolutions/apina#18]
Support converting Java enums without properties to TypeScript Enums.

**Comment by terohe** [comment]
Question: Does this feature require the backend to use Java enum (de)serialization as ordinals (instead of enum name)? `SerializationFeature.WRITE_ENUMS_USING_INDEX` and not `SerializationFeature.WRITE_ENUMS_USING_TO_STRING`?

At first sight does not seem to work with String enum values. Or am I missing something? :)

**Comment by komu** [comment]
No, you should be using strings on the wire. How are you representing enums in TypeScript-code?

Are you aware that TypeScript internally requires enums to be represented as numbers instead of strings? Apina's serialization then converts the internal number representation to String using code like `MyEnum[originalValue]`, which maps it to corresponding number. The catch is that `MyEnum[originalValue]` also works the other way: if `originalValue` is a string, it will get converted to a number.

Anyway, if you say `MyEnum.MY_CONSTANT` in TypeScript code, the runtime value will be a number instead of string "MY_CONSTANT". Perhaps your original value is coming from a data-bound string-attribute or something like that and the value is already garbage before Apina even sees it?

Anyway, the whole enum-values are really numbers is quite inconvenient many times, so personally I've ditched TypeScript enums in favor of string union types:

```TypeScript
type MyEnum = "FIRST"  | "SECOND" | "THIRD";
```

This way TypeScript code and HTML templates can all work with same representation, but still get type checking.

In the future I'll add direct support for Apina to build enums like this, but for the time being you can define this in your own file and then instruct Apina to use your own type:

```groovy
apina {
    target = new File(project(":scheduling-frontend").projectDir, 'src/app/apina.ts')
    imports = ['./my-apina-types': ['MyEnum']]
}
```

Now Apina will not attempt to generate type for `MyEnum`, but will import your type. Finally, you'll need to register a serializer, but since strings can be used as it is, simply `apinaConfig.registerIdentitySerializer('MyEnum')` will do.

**Comment by terohe** [comment]
Ach so! Thanks for the really quick and thorough reply! The data-bound strings in HTML and such seem to cause the confusion indeed (string -> number). Just started to get the hang of it when you replied.

And thanks for the tip! We'll consider using the string unions instead of enums. However, that would mean duplicating the enums (bummer).

**Comment by komu** [comment]
I'll try to add the option for automatically generating String-based enums soon.

**Comment by komu** [comment]
Just pushed 72b3d3fe5d9d8083d6e02b993878c589d97442c9 to implement this feature. I'll make a new release after a bit of testing and documentation.

**Comment by komu** [comment]
I improved [README.md](https://github.com/EvidentSolutions/apina/blob/master/README.md) and released `0.7.0` with this feature.
